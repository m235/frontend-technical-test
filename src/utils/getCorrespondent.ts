import type { Conversation } from '@/types/conversation'

export const getCorrespondent = (userId: number, conversation: Conversation) => {
  switch (userId) {
    case conversation.senderId:
      return conversation.recipientNickname
    case conversation.recipientId:
      return conversation.senderNickname
    default:
      throw new Error('You are nor the recipient nor the sender')
  }
}
