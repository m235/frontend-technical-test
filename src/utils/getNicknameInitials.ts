export const getNicknameInitials = (nickname: string) =>
  nickname
    .split(' ')
    .map((string) => string.trim().charAt(0))
    .slice(0, 2)
    .join('')
