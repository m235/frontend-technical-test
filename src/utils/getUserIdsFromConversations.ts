import type { Conversation } from '@/types/conversation'

export const getUserIdsFromConversations = (conversations?: Conversation[]) =>
  Array.from<number>(new Set(conversations?.reduce((acc, curr) => [...acc, curr.recipientId, curr.senderId], []) ?? []))
