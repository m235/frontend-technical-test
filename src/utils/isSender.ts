import type { Conversation } from '@/types/conversation'

export const isSender = (userId: number, conversation: Conversation) => userId === conversation.senderId
