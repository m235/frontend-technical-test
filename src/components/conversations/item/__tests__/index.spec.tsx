import ConversationListItem from '@/components/conversations/item'
import useAuth from '@/hooks/useAuth'
import { render } from '@/tests/utils'
import type { Conversation } from '@/types/conversation'

const mockedUseAuth = useAuth as jest.Mock<any>
jest.mock('@/hooks/useAuth')

describe('components:conversations:item', () => {
  const conversation: Conversation = { id: 1, recipientNickname: 'John', recipientId: 1, senderNickname: 'Travolta', senderId: 2 }

  beforeEach(() => {
    mockedUseAuth.mockImplementation(() => ({ userId: 1, user: [{ id: 1, nickname: 'John Travolta', token: 'x' }] }))
  })
  afterEach(() => {
    jest.clearAllMocks()
  })
  it('should render default correctly', () => {
    const { container } = render(<ConversationListItem conversation={conversation} />)

    expect(container).toMatchSnapshot()
  })
})
