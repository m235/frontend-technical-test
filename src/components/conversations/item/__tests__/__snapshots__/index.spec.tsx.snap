// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components:conversations:item should render default correctly 1`] = `
<div>
  <a
    href="/messages/1"
  >
    <div
      class="sc-eCYdqJ isacuL"
      role="listitem"
    >
      <div
        class="sc-hKMtZM faynUP sc-jSMfEi bMewtJ"
      >
        T
      </div>
      <div
        class="sc-gKXOVf gjdvix"
      >
        <h2
          class="sc-iBkjds bjHoco"
        >
          <span>
            Travolta
          </span>
        </h2>
        <span
          class="sc-bczRLJ eockqr"
          data-testid="loading"
        />
      </div>
    </div>
  </a>
</div>
`;
